### ISSUE TITLE
Error on Repo List (invalid_type at .List.Repos.5.Repository.Branches (expected array))

### ISSUE THREAD

**Issue by MarcWort** [isker/neogrok#30]
Hi,

neogrok seems to be currently incompatible with current zoekt from master (current at 27429a1e828f23ab3c97f64f318c11d984d24f73). When I click on "Repositories" neogrok shows this:

```
zoekt call failed: ValitaError: invalid_type at .List.Repos.5.Repository.Branches (expected array) (+ 6 other issues)
```
I see this in the logs:
```
neogrok-frontend-1    | zoekt call failed: ValitaError: invalid_type at .List.Repos.5.Repository.Branches (expected array) (+ 6 other issues)
neogrok-frontend-1    |     at ObjectType.parse (file:///app/node_modules/@badrap/valita/dist/node-mjs/index.mjs:275:19)
neogrok-frontend-1    |     at listRepositories (file:///app/build/server/chunks/zoekt-list-repositories-yabmNVcH.js:22:31)
neogrok-frontend-1    |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
neogrok-frontend-1    |     at async file:///app/build/server/chunks/devalue-bypass-7oEQcv2_.js:13:26
neogrok-frontend-1    |     at async render_endpoint (file:///app/build/server/index.js:1511:20)
neogrok-frontend-1    |     at async resolve2 (file:///app/build/server/index.js:4033:22)
neogrok-frontend-1    |     at async Object.handle (file:///app/build/server/chunks/hooks.server-cmBqbttp.js:15:22)
neogrok-frontend-1    |     at async respond (file:///app/build/server/index.js:3927:22)
neogrok-frontend-1    |     at async fetch (file:///app/build/server/index.js:3710:26) {
neogrok-frontend-1    |   issueTree: {
neogrok-frontend-1    |     ok: false,
neogrok-frontend-1    |     code: 'prepend',
neogrok-frontend-1    |     key: 'List',
neogrok-frontend-1    |     tree: { ok: false, code: 'prepend', key: 'Repos', tree: [Object] }
neogrok-frontend-1    |   },
neogrok-frontend-1    |   _issues: undefined
neogrok-frontend-1    | }
```

Searching works fine.

**Comment by isker** [comment]
Thanks for the report. I'll take a look.

**Comment by MarcWort** [comment]
Some more informations: putting in some search query on the repository site like "a" neogrok shows `SyntaxError: Unexpected token 'C', "Cross-site"... is not valid JSON
` if I than reload the page it works.

Directly opening `http://localhost:8081/repositories?q=a` also works.

**Comment by isker** [comment]
Somehow we are running afoul of [SvelteKit's CSRF prevention](https://svelte.dev/docs/kit/adapter-node#Environment-variables) when I don't think this used to happen.  You can fix it by setting the ORIGIN environment variable when running neogrok, telling it what site neogrok is running on.  For example, this works:
```
λ ORIGIN='http://localhost:3000' ZOEKT_URL='http://localhost:6070' neogrok
Listening on 0.0.0.0:3000
``` 

I will leave this issue open while I try to understand what changed and make any needed fixes.

**Comment by MarcWort** [comment]
You are right, thank you. This fixes the repository list after typing something in the searchbar (`Unexpected token 'C'` error). 

But without typing something it is still `zoekt call failed: ValitaError: invalid_type at .List.Repos.21.Repository.Branches (expected array) (+ 6 other issues)` 

TLDR: `http://localhost:8081/repositories` does not work, but `http://localhost:8081/repositories?q=a` does.

**Comment by MarcWort** [comment]
![image](https://github.com/user-attachments/assets/53751a08-7328-43db-8c5f-94083232f529)

![image](https://github.com/user-attachments/assets/1daa31f4-ebab-4355-be73-2ea1741c67e8)

**Comment by isker** [comment]
I believe I have found fixes for both of your bugs. They were two completely unrelated problems.

I've pushed the fixes and will publish a release on npm shortly. Let me know how it goes for you.

**Comment by isker** [comment]
1.0.2 is published.

**Comment by MarcWort** [comment]
I just wanted to say thanks for the two quick fixes. Everything's working fine now.
